### ISSUE TITLE
Get "NOSCRIPT No matching script. Please use EVAL" when redis pod restarts.

### ISSUE THREAD

**Issue by WanderingTurtle** [express-rate-limit/rate-limit-redis#144]
### Description

Please consider adding a try catch around https://github.com/wyattjoh/rate-limit-redis/blob/effebd57bf943d3acfb3cdf3c4856c72383cd749/src/lib.ts#L114 and https://github.com/wyattjoh/rate-limit-redis/blob/effebd57bf943d3acfb3cdf3c4856c72383cd749/src/lib.ts#L121, so that when RedisNoScriptException happens, loadScript() can be invoked again to load the script into redis.

Current workaround:
```
const app = express()
class MyRedisStore extends RedisStore { 
    increment = async(key) => {
        let results = []
        try {
            results = await this.sendCommand(
                "EVALSHA",
                await this.loadedScriptSha1,
                "1",
                this.prefixKey(key),
                this.resetExpiryOnChange ? "1" : "0",
                this.windowMs.toString()
            );
        } catch (err) {
            this.loadedScriptSha1 = await this.loadScript()
            results = await this.sendCommand(
                "EVALSHA",
                await this.loadedScriptSha1,
                "1",
                this.prefixKey(key),
                this.resetExpiryOnChange ? "1" : "0",
                this.windowMs.toString()
            );
        }
      
        if (!Array.isArray(results)) {
            throw new TypeError("Expected result to be array of values");
        }
    
        if (results.length !== 2) {
            throw new Error(`Expected 2 replies, got ${results.length}`);
        }
    
        const totalHits = results[0];
        if (typeof totalHits !== "number") {
            throw new TypeError("Expected value to be a number");
        }
    
        const timeToExpire = results[1];
        if (typeof timeToExpire !== "number") {
            throw new TypeError("Expected value to be a number");
        }
    
        const resetTime = new Date(Date.now() + timeToExpire);
        return {
            totalHits,
            resetTime,
        };
    }
}

function apiLimiterDefault() {
    return rateLimit({
        ........
        store: new MyRedisStore({
            sendCommand: (...args) => xdrRedis.boot().handle().sendCommand(args),
        })
    })
}

app.get("/", [apiLimiterDefault()], (req, res) {
  return res.json({"hello": "world"})
})
```
(Basically just wrapping a try catch block around)

### Library version

3.0.1

### Node version

14.16.0

### Module system

CommonJS

### Redis Client

redis 4.0.4

**Comment by youssefm** [comment]
We're also seeing this error in production, would love to see this fixed.

**Comment by andersontechnetverrum** [comment]
When are we releasing this?

**Comment by mifi** [comment]
create a pr #160

**Comment by antoniojps** [comment]
I'm also seeing this issue! Any plan to merge @mifi PR?

**Comment by elizatlawy** [comment]
Any update on this?

**Comment by phuze** [comment]
Any update on this?

**Comment by elizatlawy** [comment]
@phuze Thanks, that worked.
Just a small correction the name of the package is` rate-limit-redis` so you should run: `npx patch-package rate-limit-redis`
or: `yarn patch-package rate-limit-redis` if you are using `yarn` like me.

**Comment by mifi** [comment]
Yes, someone needs to review the pr

**Comment by Garett-MacGowan** [comment]
+1

**Comment by gamemaker1** [comment]
Hi everyone,

Apologies for the delay in solving this issue, and thank you so much for all your help.

The issue should be fixed in [v3.1.0](https://github.com/express-rate-limit/rate-limit-redis/releases/tag/v3.1.0) thanks to @mifi's PR.
